Lombiq.Hosting.Tenants.FeaturesGuard/Readme.md

```diff
@@ -4,37 +4,42 @@
 
 ## About
 
-Module to prevent disabling a configurable set of features on tenants.
+A module that makes it possible to conditionally enable and conditionally keep enabled, or entirely prevent enabling, configurable sets of features on tenants.
 
 ## Documentation
 
-- To use this feature, enable it on both the Default and the user tenant.
-- Once enabled on the user tenant, the Features Guard feature cannot be disabled.
-- Features that should not be deactivatable can be specified in the appsettings JSON file using `AlwaysEnabledFeaturesOptions`. Example configuration:
+### Conditionally enabling features
+
+- To use this feature, enable it on both the Default and the user tenant, and make sure the app is hosted on Azure.
+- Features that should be conditionally enabled, as well as the features whose status acts as the condition, can be specified in appsettings.json using `ConditionallyEnabledFeaturesOptions`.
+Conditionally enabled features need to be provided with a singular or multiple condition features, where the status of the condition features determines whether the corresponding conditional feature is enabled or disabled. Example configuration:
 
 ```json
 {
   "OrchardCore": {
     "Lombiq_Hosting_Tenants_FeaturesGuard": {
-      "AlwaysEnabledFeaturesOptions": {
-        "AlwaysEnabledFeatures": [
-          "OrchardCore.Roles",
-          "OrchardCore.Users"
-        ]
+      "ConditionallyEnabledFeaturesOptions": {
+        "ConditionallyEnabledFeatures": {
+          "EnableFeatureIfOtherFeatureIsEnabled": {
+            // "Enable this feature and keep it enabled": "If any of these features are enabled",
+            "OrchardCore.Media.Azure.Storage": "OrchardCore.Media",
+            "OrchardCore.Twitter": "OrchardCore.Media, OrchardCore.Workflows"
+          }
+        }
       }
     },
-  }
 }
 ```
 
-- Additionally, the feature ensures whenever the OrchardCore.Media feature is enabled, the following features also get enabled:
-  - OrchardCore.Content.Types
-  - OrchardCore.Liquid
-  - OrchardCore.Media.Azure.Storage
-  - OrchardCore.Media.Cache
-  - OrchardCore.Settings
+Example case 1:
+- Enables `Azure Storage` when `Media` is enabled. Keeps `Azure Storage` enabled as long as `Media` is enabled, with the exception where if one of `Azure Storage`'s dependencies is disabled, `Azure Media` does not get re-enabled.
```

Comment:
In this example, Azure Storage shouldn't be disabled in any case, including if any of its dependencies get disabled, as long as Media is enabled. Otherwise, people can easily defeat this.

Reply:
Artist's depiction of users defeeting FeaturesGuard:
![image](https://user-images.githubusercontent.com/51411356/199022901-9471b3b0-9f7a-483b-81bd-3dfbafae7fba.png)

Reply:
:D * insert comic of soldier defending sleeping person, but badly *

Reply:
![image](https://user-images.githubusercontent.com/51411356/199192412-294246c2-c5c4-42db-93cb-e415772861f3.png)

Reply:
Thank you :D.

Reply:
I think, you should pursue a comic writer career - at least on the side, @porgabi.